services/api/lib/bulk.js

```diff
@@ -152,21 +152,44 @@ exports.register = function(server, options, done) {
   // 0th index and the value on that object keyed with 'id'
   var reachRegex = /^\$(\d+)\.(.*)$/;
 
+  var pipelineKeys = [
+    'id',
+    'projectId',
+    'pageId'
+  ];
+
+  function getIdKey(data) {
+    for (var i = 0; i < pipelineKeys.length; i++) {
+      if (data[pipelineKeys[i]]) {
+        return pipelineKeys[i];
+      }
+    }
+  }
+
+  function isReachString(value) {
+    return reachRegex.test(value);
+  }
+
+  function replaceValue(data, key, actionIndex, txResults) {
```

Comment:
this function makes me uneasy-- it takes in certain parameters, does a replace, and returns a side effect(?) of a function that is called during a replace on a string that... doesn't get used again?  maybe i am misunderstanding but i think that we should consider rewriting it so that it better expresses the purpose of the code.